scipy/signal/signaltools.py

```diff
@@ -458,16 +512,25 @@ def convolve(in1, in2, mode='full'):
     if volume.ndim == kernel.ndim == 0:
         return volume * kernel
 
+    # see whether the fourier transform convolution method or the direct
+    # convolution method is faster (discussed in scikit-image PR #1792)
+    big_O_constant = 1 / 40.032 if kernel.ndim > 1 else 1 / 1.5
+    direct_time = big_O_constant * np.prod(volume.shape + kernel.shape)
+    fft_time = np.sum([n * np.log(n) for n in volume.shape + kernel.shape])
+
+    if ((fft_time < direct_time and method == 'auto') or method == 'fft') \
+            and volume.dtype.kind in 'buif' and kernel.dtype.kind in 'buif':
+        out = fftconvolve(volume, kernel, mode=mode)
+        out = np.around(out) if volume.dtype.kind in 'ui' else out
+        return np.asarray(out, dtype=volume.dtype)
+
     # fastpath to faster numpy 1d convolve
     if volume.ndim == 1 and kernel.ndim == 1 and volume.size >= kernel.size:
         return np.convolve(volume, kernel, mode)
 
     slice_obj = [slice(None, None, -1)] * len(kernel.shape)
```

Comment:
This one could be changed to `reverse = [slice(None, None, -1)] * kernel.ndim` also